sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala

```diff
@@ -1928,4 +1928,17 @@ class HiveDDLSuite
       }
     }
   }
+
+  test("SPARK-20680: Spark-sql do not support for void column datatype of view") {
+    withTable("t", "tabNullType") {
+      val client = spark.sharedState.externalCatalog.asInstanceOf[HiveExternalCatalog].client
+      client.runSqlHive("CREATE TABLE t (t1 int)")
+      client.runSqlHive("INSERT INTO t VALUES (3)")
+      client.runSqlHive("CREATE TABLE tabNullType AS SELECT NULL AS col FROM t")
+      assert(spark.table("tabNullType").schema.contains(StructField("col", NullType)))
```

Comment:
From the Jenkins, I saw below error message:

> org.scalatest.exceptions.TestFailedException: StructType(StructField(col,NullType,true)) did not contain StructField(col,NullType,true)

But it can passed from my spark-shell:
> scala> val schema = spark.table("tabNullType").schema
schema: org.apache.spark.sql.types.StructType = StructType(StructField(col,NullType,true))
scala> schema.contains(StructField("col", NullType))
res7: Boolean = true